sys/linux/socket_netlink_route.txt

```diff
@@ -850,6 +878,58 @@ br_mdb_entry_sub_u [
 	in6_addr	ipv6_addr
 ]
 
+rtm_nh_policy_get_id [
+	NHA_ID	nlattr[NHA_ID, int32[0:2]]
+] [varlen]
+
+rtm_nh_policy_get_dump [
+	NHA_OIF		nlattr[NHA_OIF, ifindex]
+	NHA_GROUPS	nlattr[NHA_GROUPS, void]
+	NHA_MASTER	nlattr[NHA_MASTER, int32[0:2]]
+	NHA_FDB		nlattr[NHA_FDB, void]
+] [varlen]
+
+rtm_nh_policy_new [
+	NHA_ID		nlattr[NHA_ID, int32[0:2]]
+	NHA_GROUP	nlattr[NHA_GROUP, array[nexthop_grp]]
+	NHA_GROUP_TYPE	nlattr[NHA_GROUP_TYPE, flags[nha_group_types, int16]]
+	NHA_OIF		nlattr[NHA_OIF, ifindex]
+	NHA_GATEWAY	nlattr[NHA_GATEWAY, nha_gateway]
+	NHA_ENCAP_TYPE	nlattr[NHA_ENCAP_TYPE, flags[lwtunnel_encap_types, int16]]
+	NHA_ENCAP	nlnest[NHA_ENCAP, lwtunnel_policy]
+] [varlen]
+
+rtm_nh_policy_new_fdb [
+	NHA_ID		nlattr[NHA_ID, int32[0:2]]
+	NHA_GROUP	nlattr[NHA_GROUP, array[nexthop_grp]]
+	NHA_GROUP_TYPE	nlattr[NHA_GROUP_TYPE, flags[nha_group_types, int16]]
+	NHA_GATEWAY	nlattr[NHA_GATEWAY, nha_gateway]
+	NHA_FDB		nlattr[NHA_FDB, void]
+] [varlen]
+
+rtm_nh_policy_new_blackhole [
+	NHA_ID		nlattr[NHA_ID, int32[0:2]]
+	NHA_GROUP	nlattr[NHA_GROUP, array[nexthop_grp]]
+	NHA_GROUP_TYPE	nlattr[NHA_GROUP_TYPE, flags[nha_group_types, int16]]
+	NHA_BLACKHOLE	nlattr[NHA_BLACKHOLE, void]
+] [varlen]
+
+rtm_nh_policy_del [
+	NHA_ID	nlattr[NHA_ID, int32[0:2]]
+] [varlen]
```

Comment:
Now that we removed one of the options from the union, having left with a single option, we don't need a union here anymore.

We can make it a type definition (to keep the convention of keeping policy definitions together in one place).

Reply:
You mean like `type rtm_nh_policy_del nlattr[NHA_ID, int32[0:2]]` ?
I thought of keeping the union just to keep it similar to the others, but I've added this change.